src/jit/lsraxarch.cpp

```diff
@@ -152,17 +151,11 @@ void Lowering::TreeNodeInfoInit(GenTree* tree)
 
         case GT_STORE_LCL_FLD:
         case GT_STORE_LCL_VAR:
-#ifdef _TARGET_X86_
-            if (tree->gtGetOp1()->OperGet() == GT_LONG)
-            {
-                tree->gtGetOp1()->SetContained();
-                info->srcCount = 2;
-            }
-            else
-#endif // _TARGET_X86_
-            {
-                info->srcCount = 1;
-            }
+            ContainCheckStoreLoc(tree->AsLclVarCommon());
+            // TODO-XArch-CQ: This should be moved to Lowering, but it widens the types, which changes the behavior
+            // of the containment analysis.
+            LowerStoreLoc(tree->AsLclVarCommon());
+            info->srcCount = GetOperandSourceCount(tree->gtOp.gtOp1);
             info->dstCount = 0;
             TreeNodeInfoInitStoreLoc(tree->AsLclVarCommon());
```

Comment:
OK, though there's not a lot of value in merging the `ContainCheckStoreLoc` since it's moving to `Lowering`, and the setting of `dstCount` will ultimately be done by default for all nodes based on `IsValue() && !isContained()`.